src/framework/audio/internal/fx/reverb/reverbprocessor.cpp

```diff
@@ -287,7 +287,7 @@ ReverbProcessor::ReverbProcessor(const AudioFxParams& params, audioch_t audioCha
     setParameter(Params::PreDelayMs, getParameter(Params::PreDelayMs));
     setParameter(Params::FeedbackTop, getParameter(Params::FeedbackTop));
 
-    setFormat(audioChannelsCount, 44100.0 /*sampleRate*/, 512 /*maximumBlockSize*/);
+    setFormat(audioChannelsCount, audioConfiguration()->sampleRate(), 512 /*maximumBlockSize*/);
```

Comment:
I'm not sure if this is correct. I believe audioConfiguration returns the sample rate for the current output device (although it seems that this value is never set) but during export, a different value might be used. 

Also, given `ReverbProcessor::setSampleRate `, is this necessary at all?

Reply:
> I believe audioConfiguration returns the sample rate for the current output device ...
> but during export, a different value might be used.

export has its own config setting ([export] audio\sampleRate=48000),  set by audioExportConfiguration()->setExportSampleRate(rate)


> (although it seems that this value is never set) but during export, a different value might be used.

It seems to be read from config file ([io]/samplerate), or if not existing config, set by https://github.com/musescore/MuseScore/blob/master/src/framework/audio/internal/audioconfiguration.cpp#L74

Trying to remove ```setFormat(audioChannelsCount, ...)``` from ReverbProcessor constructor will cause assertion error:

```
12:51:12.323 | ERROR | 140366201882304 | ReverbProcessor::setFormat | "ASSERT FAILED!": audioChannelsCount > 0 && audioChannelsCount <= 2, file: src/framework/audio/internal/fx/reverb/reverbprocessor.cpp, line: 529
```

Reply:
> export has its own config setting ([export] audio\sampleRate=48000), set by audioExportConfiguration()->setExportSampleRate(rate)

Yes, so my point was that during export, the value from audioexportconfiguration should be used instead. 

> Trying to remove setFormat(audioChannelsCount, ...) from ReverbProcessor constructor will cause assertion error:

Ah yeah, I didn't mean to remove it entirely but I wonder if it matters at all which sample rate you pass to this method initially, because it should be overridden anyway when setSampleRate is called.

Reply:
Setting to zero still triggers an assertion:
```
14:18:43.089 | DEBUG | 140147225659072 | FluidSynth      | Selected preset 'Standard' on channel 0
mscore: src/framework/audio/internal/fx/reverb/reverbprocessor.cpp:830: bool muse::audio::fx::ReverbProcessor::Processor::setFormat(muse::audio::audioch_t, double, int32_t): Assertion `sampleRate > 0.0' failed.
```

Regarding the other to usages of hardcoded samplerate 44100, they too would need to be dynamically set depending on whether playback or export samplerate is wanted.
So perhaps a targetSampleRate member is needed? (whereas the current sampleRate is the internal samplerate that the reverb is using when doing its calculation).

Reply:
I think we should undo this change, for two reasons:
- It has no effect, because before playback begins, the correct sample rate is set anyway by calling `setSampleRate`
- `audioConfiguration()->sampleRate()` is not a correct value, because it's only related to what the user put in their settings or what the current driver does, and does not take audio export into account. 



> So perhaps a targetSampleRate member is needed? (whereas the current sampleRate is the internal samplerate that the reverb is using when doing its calculation).

What would be the advantage of that? `ReverbProcessor::setSampleRate` already updates `m_processor._sampleRate`, so that will always match the output/target sample rate.

Something that does look suspicious to me is line -383, `float scale = float(getParameter(LateRoomScale) * m_processor._sampleRate / 44100.);`. Perhaps that should become just `float scale = float(getParameter(LateRoomScale))`. I don't know this piece of the code very well so I can't tell for sure what it does exactly, but I'd say, try it out :)